File: Programming EF Code First/chapter5/Chapter 5 VB/Fluent/DataAccess/DestinationConfiguration.vb
Imports System.Data.Entity.ModelConfiguration
Imports Model

Namespace DataAccess
  Public Class DestinationConfiguration
	  Inherits EntityTypeConfiguration(Of Destination)
	Public Sub New()
      Me.Property(Function(d) d.DestinationId).HasColumnName("LocationID")

      Me.Property(Function(d) d.Name).IsRequired().HasColumnName("LocationName")

      Me.Property(Function(d) d.Description).HasMaxLength(500)

      Me.Property(Function(d) d.Photo).HasColumnType("image")

      Me.ToTable("Locations", "baga")

      Me.Ignore(Function(d) d.TodayForecast)
	End Sub
  End Class
End Namespace

File: Programming EF Code First/chapter5/Chapter5 CSharp/BAGA_FluentAPI/DataAccess/DestinationConfiguration.cs
using System.Data.Entity.ModelConfiguration;
using Model;

namespace DataAccess
{
  public class DestinationConfiguration : EntityTypeConfiguration<Destination>
  {
    public DestinationConfiguration()
    {
      Property(d => d.DestinationId)
        .HasColumnName("LocationID");

      Property(d => d.Name)
        .IsRequired()
        .HasColumnName("LocationName");

      Property(d => d.Description)
        .HasMaxLength(500);

      Property(d => d.Photo)
        .HasColumnType("image");

      ToTable("Locations", "baga");

      Ignore(d => d.TodayForecast);
    }
  }
}